internal/signature/signature.go

```diff
@@ -36,13 +37,22 @@ func Verify(ctx context.Context, ref string, signature *index.Signature) error {
 		return nil
 	}
 
-	v := cosign.VerifyCommand{
-		CertVerifyOptions: options.CertVerifyOptions{
-			CertIdentity:         signature.Cosign.CertificateIdentity,
-			CertIdentityRegexp:   signature.Cosign.CertificateIdentityRegexp,
-			CertOidcIssuer:       signature.Cosign.CertificateOidcIssuer,
-			CertOidcIssuerRegexp: signature.Cosign.CertificateOidcIssuerRegexp,
-		},
+	if signature.Cosign.KeyRef == "" {
```

Comment:
I think this does not need `if and elses` can be just the entire thing as was before, because in the case before we can only have either `CertificateIdentity` or `CertificateIdentityRegexp`

so then it will be 

```
v := cosign.VerifyCommand{
		CertVerifyOptions: options.CertVerifyOptions{
			CertIdentity:         signature.Cosign.CertificateIdentity,
			CertIdentityRegexp:   signature.Cosign.CertificateIdentityRegexp,
			CertOidcIssuer:       signature.Cosign.CertificateOidcIssuer,
			CertOidcIssuerRegexp: signature.Cosign.CertificateOidcIssuerRegexp,
                         // used when using the kms verify
			KeyRef:     signature.Cosign.KeyRef,
			IgnoreTlog: signature.Cosign.PrivateInfrastructure,
		},
}
```


I don't have a KMS to test, but is it possible for you to check for both cases? I think this will work, and it simplifies a lot.


Thanks for this PR